site/en/posts/android-package-signer.md

```diff
@@ -0,0 +1,87 @@
+
+## Android Package Signer and Bubblewrap
+
+[Bubblewrap](https://github.com/GoogleChromeLabs/bubblewrap) is a project created to help web developers package their Progressive Web Apps for distribution on Google Play and has been used to be the foundation of app bundling services. Often, though, this meant users were handing over the Android signing keys and password when creating a new project without knowing if they were being stored or potentially leaked. Having a leaked key could allow a malicious party to impersonate a developer and release harmful package updates that would otherwise appear to be legitimate.
+
+## Installation and Usage
+
+The [Android Package Signer repository](https://github.com/chromeos/android-package-sign-js) is the best place to go to keep up-to-date with the package.
+
+To add Android Package Signer to your project, first install it from NPM:
+
+````bash {title="bash" .code-figure}
+npm i @chromeos/android-package-signer
+``` From there, require it in your project and initialize it:
+
+```typescript {title="Typescript" .code-figure}
+import { PackageSigner } from 'android-package-signer';
+const packageSigner = new PackageSigner(password: string, alias: string = 'android');
+````
+
+The password is a string and should be a minimum of six characters long. This will protect your keystore, so the longer the password, the better.
+
+This will instantiate a class that can be used to generate a key and sign a package with a key. To generate a key, pass the class's `generateKey` method a DName object, structured below.
+
+```typescript {title="Typescript" .code-figure}
+export interface DName {
+  commonName: string;
+  organizationName: string;
+  organizationUnit: string;
+  countryCode: string;
+}
+
+// In your code
+import { PackageSigner } from '@chromeos/android-package-signer';
+async function keyGen(): Promise<string> {
+  const packageSigner = new PackageSigner(password, alias);
+  const base64Der = await packageSigner.generateKey({
+    commonName: 'Alexander Nohe',
+    organizationName: 'Google, Inc',
+    organizationUnit: 'DevRel',
+    countryCode: 'US',
+  });
+
+  // To download the keys.
+  const downloadElement: HTMLAnchorElement = document.querySelector('#key-gen-results');
+  downloadElement.href = base64Der;
+  downloadElement.download = 'generatedKey.p12';
+  downloadElement.innerText = 'Download Generated Key';
+}
+```
+
+The response from the generateKey function is a base64-encoded der formatted PKCS12 keystore. To save this keystore to a file, download the base64Der string contents to a file. In the above example we use an anchor element with a href attribute containing the base64 encoded keystore.
```

Comment:
If it's possible to use the [File System Access API](https://web.dev/file-system-access/#write-file) instead of base64 encoding to an anchor's `href`, that'd be great, and provide good synergy between this use-case and and that one.

Reply:
I also think this description should come before the code block showing the example to break up the different steps more, especially because this isn't strictly part of the functionality of the library